planner/optimize.go

```diff
@@ -63,6 +64,23 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 	return optimize(ctx, sctx, node, is)
 }
 
+// OptimizeExecCached to optimize execute statement with plan cached
+func OptimizeExecCached(ctx context.Context, sctx sessionctx.Context,
```

Comment:
The function seems has nothing to do with `Cached`.

Reply:
`OptimizeExecCached` is called for when cache hit from "prepared plan cache" or "prepare.CachedPlan",  so in this function we will do `optimize` things with knowledge of this possible being resued plan.  This path skips some unnecessary steps for "execute" statement.
 Change this from to `OptimizeExecStmt` ?

Reply:
I think it can be used for non-cached prepared statement too.

Reply:
Seems we can move this function to `plannercore `